rust/connlib/shared/src/error.rs

```diff
@@ -173,6 +174,16 @@ pub enum ConnlibError {
     WriteResolvConfBackup(std::io::Error),
     #[error("Failed to rewrite `resolv.conf`: {0}")]
     RewriteResolvConf(std::io::Error),
+
+    // Error variants for `systemd-resolved` DNS control
+    #[error("`resolvectl dns` should have run: {0}")]
+    ResolvectlDnsDidntRun(std::io::Error),
+    #[error("`resolvectl dns` should have succeeded: {0}")]
+    ResolvectlDnsFailed(std::process::ExitStatus),
+    #[error("`resolvectl domain` should have run: {0}")]
+    ResolvectlDomainDidntRun(std::io::Error),
+    #[error("`resolvectl domain` should have succeeded: {0}")]
+    ResolvectlDomainFailed(std::process::ExitStatus),
```

Comment:
I think normally we don't need so much detail on what failed, we can just unify this in a single error